src/com/facebook/buck/ide/intellij/BaseIjModuleRule.java

```diff
@@ -118,18 +151,110 @@ private void addDepsAndFolder(
     }
   }
 
+  private void addDepsAndFolder(
+      IJFolderFactory folderFactory,
+      DependencyType dependencyType,
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context) {
+    addDepsAndFolder(
+        folderFactory, dependencyType, targetNode, wantsPackagePrefix, context, ImmutableSet.of());
+  }
+
+  protected void addDepsAndSources(
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context,
+      ImmutableSet<Path> resourcePaths) {
+    addDepsAndFolder(
+        SourceFolder.FACTORY,
+        DependencyType.PROD,
+        targetNode,
+        wantsPackagePrefix,
+        context,
+        resourcePaths);
+  }
+
   protected void addDepsAndSources(
       TargetNode<T, ?> targetNode, boolean wantsPackagePrefix, ModuleBuildContext context) {
     addDepsAndFolder(
         SourceFolder.FACTORY, DependencyType.PROD, targetNode, wantsPackagePrefix, context);
   }
 
   protected void addDepsAndTestSources(
+      TargetNode<T, ?> targetNode,
+      boolean wantsPackagePrefix,
+      ModuleBuildContext context,
+      ImmutableSet<Path> resourcePaths) {
+    addDepsAndFolder(
+        TestFolder.FACTORY,
+        DependencyType.TEST,
+        targetNode,
+        wantsPackagePrefix,
+        context,
+        resourcePaths);
+  }
+
+  protected void addDepsAndTestSources(
       TargetNode<T, ?> targetNode, boolean wantsPackagePrefix, ModuleBuildContext context) {
     addDepsAndFolder(
         TestFolder.FACTORY, DependencyType.TEST, targetNode, wantsPackagePrefix, context);
   }
 
+  protected ImmutableSet<Path> getResourcePaths(Collection<SourcePath> resources) {
+    return resources
+        .stream()
+        .filter(PathSourcePath.class::isInstance)
+        .map(PathSourcePath.class::cast)
+        .map(PathSourcePath::getRelativePath)
+        .collect(MoreCollectors.toImmutableSet());
+  }
+
+  protected ImmutableSet<Path> getResourcePaths(
+      Collection<SourcePath> resources, Path resourcesRoot) {
+    return resources
+        .stream()
+        .filter(PathSourcePath.class::isInstance)
+        .map(PathSourcePath.class::cast)
+        .map(PathSourcePath::getRelativePath)
+        .filter(path -> path.startsWith(resourcesRoot))
+        .collect(MoreCollectors.toImmutableSet());
+  }
+
+  protected ImmutableMultimap<Path, Path> getResourcesRootsToResources(
+      ImmutableSet<Path> resourcePaths) {
+    final JavaPackageFinder packageFinder =
+        projectConfig.getJavaBuckConfig().createDefaultJavaPackageFinder();
+    return resourcePaths
+        .stream()
+        .collect(
+            MoreCollectors.toImmutableMultimap(
+                path ->
+                    MorePaths.stripCommonSuffix(
+                            path.getParent(), packageFinder.findJavaPackageFolder(path))
```

Comment:
I doubt `src_root` is used to detect the relative location of resources in the final jar. That's the purpose of `resources_root`, right? If `resources_root` not present the resources are placed in the same location in a jar as they are relatively to target's base path. I think this logic should be reproduced here, otherwise IntelliJ and buck will have resources in different locations.

Reply:
Actually, based on https://buckbuild.com/rule/java_library.html (and on my own testing), if resources_root is not set, then it basically uses the corresponding src_root as a resources_root